micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -712,11 +711,10 @@ private boolean unmarkStaleId(Id originalId) {
         return !stalePreFilterIds.isEmpty() && stalePreFilterIds.remove(originalId);
     }
 
-    private void checkAndWarnAboutGaugeDoubleRegistration(Meter meter) {
-        if (meter instanceof Gauge) {
-            gaugeDoubleRegistrationLogger.log(() -> String.format(
-                    "This Gauge has been already registered (%s), the Gauge registration will be ignored.",
-                    meter.getId()));
+    private void checkAndWarnAboutDoubleRegistration(Meter meter) {
+        if (meter instanceof Gauge || meter instanceof FunctionCounter || meter instanceof FunctionTimer) {
+            doubleRegistrationLogger.log(() -> String.format(
+                    "This Meter has been already registered (%s), the registration will be ignored.", meter.getId()));
```

Comment:
It would be kind of nice to preserve the type in the log message, so it is more clear this doesn't apply to all types.

Reply:
I added logic to handle that. There is an interesting aspect though: `TimeGauge` is a `Gauge` so the warning will say `This Gauge has been already registered` (as it said before this change too). Not sure we should "fix" this though, what do you think?

Reply:
I think that's fine since it is existing behavior.